romsel_dsimenutheme/arm9/source/graphics/graphics.cpp

```diff
@@ -1091,47 +1127,38 @@ void topBgLoad() {
 			for (int y=15; y>=0; y--) {
 				u16 buffer[512];
 				fread(buffer, 2, 0x200, file);
-				u16* src = buffer+(small_font_texcoords[0+(4*usernameRendered[c])]);
-				for (int i=0; i<small_font_texcoords[2+(4*usernameRendered[c])]; i++) {
+				u16* src = buffer+(top_font_texcoords[0+(4*charIndex)]);
+
+				for (int i=0; i < top_font_texcoords[2+(4*charIndex)]; i++) {
 					u16 val = *(src++);
+					u16 bg = BG_GFX_SUB[(y+1)*256+(i+x)]; //grab the background pixel
+					// Apply palette here.
+					
+					// Magic numbers were found by dumping val to stdout
+					// on case default.
 					switch (val) {
+						// #ff00ff
 						case 0xFC1F:
-						default:
 							break;
+						// #414141
 						case 0xA108:
 							val = bmpPal_topSmallFont[1+((PersonalData->theme)*16)];
 							break;
-						case 0xA96A:
-							val = bmpPal_topSmallFont[2+((PersonalData->theme)*16)];
-							break;
-						case 0xB5AD:
-							val = bmpPal_topSmallFont[3+((PersonalData->theme)*16)];
-							break;
-						case 0xBDEF:
-							val = bmpPal_topSmallFont[4+((PersonalData->theme)*16)];
-							break;
-						case 0xC651:
-							val = bmpPal_topSmallFont[5+((PersonalData->theme)*16)];
-							break;
-						case 0xD294:
-							val = bmpPal_topSmallFont[6+((PersonalData->theme)*16)];
-							break;
-						case 0xDAD6:
-							val = bmpPal_topSmallFont[7+((PersonalData->theme)*16)];
+						case 0xC210:
+							// blend the colors with the background to make it look better.
+							val = alphablend(bmpPal_topSmallFont[2+((PersonalData->theme)*16)], bg, 64);
```

Comment:
If alpha is set to 32 here (`alphablend(bmpPal_topSmallFont[2+((PersonalData->theme)*16)], bg, 32);), it reduces the "bold" look, but then causes the edges to look too dark on the dark theme.